frontend/src/routes/train.jsx

```diff
@@ -0,0 +1,151 @@
+import { createEffect, createSignal } from "solid-js";
+import { createStore } from "solid-js/store";
+import { styled } from "solid-styled-components";
+import Button from "../components/form/button";
+import InputEmail from "../components/form/email-input";
+import InputHash from "../components/form/hash-input";
+import Slider from "../components/slider";
+import InputFile from "../components/form/file-input";
+import { getFilesUrls } from "../helpers/getFilesUrl";
+import { validateEmail } from "../helpers/validation";
+
+const Train = () => {
+  // store of data from inputs of form
+  const [formData, setFormData] = createStore({
```

Comment:
Should we try the [solid-forms](https://github.com/jorroll/solid-forms) library?

Reply:
examples

without nested
https://stackblitz.com/edit/vitejs-vite-lzcbdd?file=src%2FExample.tsx

with nested (same as us)
https://stackblitz.com/edit/vitejs-vite-mmmzfj?file=src%2FExample.tsx